### PR TITLE
Ubuntu base image with proxy using webpack-dev-server

### PR DESCRIPTION
**Pull request by promehul**
###### Explanation About What Code Achieves:

  - Explanation

###### Screenshots/GIFs:

  - Screenshot

###### Steps To Test:

  - Steps

###### TODOs:

  - None

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.*
```
Comment (achillesrasquinha) [inline .dockerignore:1]: don't use this. this excludes `.env` files as well.

Comment (achillesrasquinha) [reply .dockerignore:1]: add `.git`, others, etc.

File: Dockerfile [hunk 2]
```diff
@@ -15,57 +15,53 @@ RUN yarn install --pure-lockfile
 COPY . /app
 
 # Run the yarn run build script
-RUN yarn build
+CMD ["yarn", "build"]
```
Comment (achillesrasquinha) [inline Dockerfile:18]: Please explain why? Shouldn't it be `RUN`?

File: Dockerfile [hunk 3]
```diff
@@ -15,57 +15,53 @@ RUN yarn install --pure-lockfile
...
 
-# Base image that is derived from alpine and has R installed as a shared library
-FROM achillesrasquinha/rpy2
+# Use ubuntu as base image
+FROM  ubuntu:xenial
```
Comment (achillesrasquinha) [inline Dockerfile:21]: Use the smallest? If `xenial`, resolve it.

File: Dockerfile [hunk 4]
```diff
@@ -15,57 +15,53 @@ RUN yarn install --pure-lockfile
...
 
-# Add label for metadata
+# Add labels for metadata
 LABEL maintainer achillesrasquinha@gmail.com
```
Comment (achillesrasquinha) [inline Dockerfile:24]: `maintainer=achillesrasquinha@gmail.com`, better to squash all in a single layer when multiple labels.

File: Dockerfile [hunk 5]
```diff
@@ -15,57 +15,53 @@ RUN yarn install --pure-lockfile
...
-ENV JAVA_HOME=/usr/lib/jvm/java-1.8-openjdk \
-    PATH=$PATH:/usr/lib/jvm/java-1.8-openjdk/jre/bin:/usr/lib/jvm/java-1.8-openjdk/bin \
-    JAVA_VERSION=8u212 \
-    JAVA_ALPINE_VERSION=8.212.04-r0
-
-RUN set -x \
-	&& apk add --no-cache \
-	openjdk8="$JAVA_ALPINE_VERSION" \
-	&& [ "$JAVA_HOME" = "$(docker-java-home)" ]
+RUN wget https://d3pxv6yz143wms.cloudfront.net/8.212.04.2/java-1.8.0-amazon-corretto-jdk_8.212.04-2_amd64.deb && \
```
Comment (achillesrasquinha) [inline Dockerfile:40]: Wrap this in a single `RUN` Layer?

File: Dockerfile [hunk 6]
```diff
@@ -15,57 +15,53 @@ RUN yarn install --pure-lockfile
...
+    apt-get update &&  apt-get install java-common && apt-get install -y --no-install-recommends apt-utils && \
+    dpkg --install java-1.8.0-amazon-corretto-jdk_8.212.04-2_amd64.deb
+
+# Install R
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E084DAB9 \
```
Comment (achillesrasquinha) [inline Dockerfile:45]: This too.

File: Dockerfile [hunk 7]
```diff
@@ -15,57 +15,53 @@ RUN yarn install --pure-lockfile
...
+
+# Run setup.R script to install required R packages
+RUN cd /R \
+    && Rscript setup.R
 
 # Copy the pip requirements file into the container
 COPY ./requirements.txt  /requirements.txt
 
 # Install pip packages
 RUN pip3 install --upgrade pip \
```
Comment (achillesrasquinha) [inline Dockerfile:61]: Use `pip` or `pip3` everywhere.

File: webpack.config.js [hunk 8]
```diff
@@ -21,16 +21,18 @@ module.exports     = {
     filename: 'bundle.min.js',
     publicPath: 'http://' + config.host + ':' + config.port
   },
+  devServer: {
+    port: 9000,
+    proxy: {
```
Comment (achillesrasquinha) [inline webpack.config.js:26]: Does this solve the auto-reload?

File: webpack.config.js [hunk 9]
```diff
@@ -21,16 +21,18 @@ module.exports     = {
...
+      '/api': '0.0.0.0:5000',
```
Comment (achillesrasquinha) [inline webpack.config.js:27]: config.host, config.port

Comment (achillesrasquinha) [reply webpack.config.js:27]: even better, process.env now that you've got docker ;)

**Review by achillesrasquinha — APPROVED** [review]

File: .dockerignore (hunk 1 above)
Comment (promehul) [reply .dockerignore]: Where is .env file?

**Pull request closed without merging**